File: Software/Java/RabbitMQClient/src/main/java/com/mathworks/messaging/utilities/ExchangeProperties.java
package com.mathworks.messaging.utilities;

import java.util.HashMap;

public class ExchangeProperties {
    private String name;
    private String type = "topic";
    private boolean create  = true;
    private boolean durable = true;
    private boolean autoDelete = false;
    private boolean internal = false;
    private HashMap<String,Object> arguments = new HashMap<String,Object>();
    
    public HashMap<String, Object> getArguments() {
        return arguments;
    }
    public void setArguments(HashMap<String, Object> arguments) {
        this.arguments = arguments;
    }
    public boolean isCreate() {
        return create;
    }
    public void setCreate(boolean create) {
        this.create = create;
    }
    
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }
    public boolean isDurable() {
        return durable;
    }
    public void setDurable(boolean durable) {
        this.durable = durable;
    }
    public boolean isAutoDelete() {
        return autoDelete;
    }
    public void setAutoDelete(boolean autoDelete) {
        this.autoDelete = autoDelete;
    }
    public boolean isInternal() {
        return internal;
    }
    public void setInternal(boolean internal) {
        this.internal = internal;
    }
}

File: Software/MATLAB/app/system/+rabbitmq/ExchangeProperties.m
classdef ExchangeProperties < rabbitmq.object
    % EXCHANGEPROPERTIES Exhange related configuration used in
    % rabbitmq.ConnectorProperties.
    %
    % EXCHANGEPROPERTIES Properties:
    %
    %   name - Name of the exchange
    %          Type: string
    %          Default: "amq.topic"
    %
    %   type - Type of exchange
    %          Type: string
    %          Default: "topic"
    %
    %   create - Ensure exchange exists
    %            Type: logical
    %            Default: true
    %
    %   durable - Work with a durable exchange
    %             Type: logical
    %             Default: true
    %
    %   autoDelete - Work with an auto delete exchange
    %                Type: logical
    %                Default: false
    %
    %   internal - Work with an internal exchange
    %              Type: logical
    %              Default: false
    %
    % See Also CONNECTORPROPERTIES

    % Copyright 2023 The MathWorks, Inc.
    properties
        name
        type
        create
        durable
        autoDelete
        internal
    end
    properties (Dependent)
        arguments
    end    
    methods
        function obj = ExchangeProperties(options)
            arguments
                options.name string = "amq.topic"
                options.type string = "topic"
                options.create logical = true
                options.durable logical = true
                options.autoDelete logical = false
                options.internal logical = false                
            end
            obj.Handle = com.mathworks.messaging.utilities.ExchangeProperties();
            for p = string(fieldnames(options))'
                obj.(p) = options.(p);
            end
        end
        function set.name(obj,val)
            obj.Handle.setName(val);
            obj.name = val;
        end
        function set.type(obj,val)
            obj.Handle.setType(val);
            obj.type = val;
        end        
        function set.create(obj,val)
            obj.Handle.setCreate(val);
            obj.create = val;
        end
        function set.durable(obj,val)
            obj.Handle.setDurable(val);
            obj.durable = val;
        end
        function set.internal(obj,val)
            obj.Handle.setInternal(val);
            obj.internal = val;
        end
        function set.autoDelete(obj,val)
            obj.Handle.setAutoDelete(val);
            obj.autoDelete = val;
        end
        function a = get.arguments(obj) 
            a = obj.Handle.getArguments();
        end
    end
end